jujugui/static/gui/src/app/components/deployment/summary/summary.js

```diff
@@ -43,6 +43,55 @@ YUI.add('deployment-summary', function() {
       validateForm: React.PropTypes.func.isRequired
     },
 
+    getInitialState: function() {
+      return {
+        regions: [],
+        activeRegion: this.props.deploymentStorage.region
+      };
+    },
+
+    componentWillMount: function() {
+      // If the model has already been deployed then we already
+      // have all the information we need.
+      var props = this.props;
+      if (props.modelCommitted) { return; }
+      // Because the user may have selected a template from the choose-cloud
+      // component we won't have any of the location data for the template
+      // to be able to properly select the correct region or cloud from the
+      // drop down list.
+      var deploymentStorage = props.deploymentStorage;
+      var templateName = deploymentStorage.templateName;
+      props.jem.listTemplates((error, credentials) => {
+        if (error) {
+          console.error('Unable to list templates', error);
+          return;
+        }
+        // The credentials are returned in an array so we need to loop through
+        // the list to find the one with the proper path.
+        credentials.forEach(credential => {
+          if (credential.path === templateName) {
+            deploymentStorage.cloud = credential.location.cloud;
+            deploymentStorage.region = credential.location.region;
+          }
+        });
+        if (deploymentStorage.cloud && deploymentStorage.region) {
+          props.jem.listRegions(deploymentStorage.cloud, (error, regions) => {
+            if (error) {
+              console.error('Unable to list regions', error);
+              return;
+            }
+            this.setState({
+              regions: regions,
+              activeRegion: deploymentStorage.region
+            });
+          });
+        } else {
+          console.log(
```

Comment:
Maybe `console.error`. We should probably even remove the ability to set the region if we don't have this info.